spec/dynamoid/persistence_spec.rb

```diff
@@ -65,6 +65,12 @@
     expect(@user.send(:dump)[:name]).to eq 'Josh'
   end
 
+  it 'keeps raw Hash attributes as Hash' do
+    config = {acres: 5, trees: {cyprus: 30, poplar: 10, joshua: 1}, horses: ['Lucky', 'Dummy'], lake: 1, tennis_court: 1}
+    @addr = Address.new(:config => config)
+    expect(@addr.send(:dump)[:config]).to eq config
+  end
+
```

Comment:
Given that we have an `if value.is_a? Hash` in `#undump_field`, could we add tests for the other data types and make sure they're sane? `Array`, `String`, `Numeric`, and `nil` would be great.